packages/hardhat-ethers/test/hardhat-ethers-provider.ts

```diff
@@ -483,6 +484,44 @@ describe("hardhat ethers provider", function () {
     });
   });
 
+  describe("issue 4970", function () {
+    it("issue 4970", async function () {
+      const userAddress = "0x5FbDB2315678afecb367f032d93F642f64180aa3";
+      const amount = toBeHex("1000000", 32);
+      const usdc = "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48";
+
+      const erc20 = new Contract(
+        usdc,
+        ["function balanceOf(uint256) public view returns (uint256)"],
+        this.env.ethers.provider
+      );
+
+      // TODO: How to get the correct storage index of `balanceOf`?
+      const balanceStorageSlot = 9;
+      const hexUserStorageSlot = AbiCoder.defaultAbiCoder().encode(
+        ["address", "uint256"],
+        [userAddress, balanceStorageSlot]
+      );
+
+      const userStorageSlot = stripZerosLeft(keccak256(hexUserStorageSlot));
+      console.log("slot:", userStorageSlot);
+
+      await this.env.network.provider.send("hardhat_setStorageAt", [
+        usdc,
+        userStorageSlot,
```

Comment:
@fvictorio this is the part that I'm unfamiliar with. How can you calculate the storage index for the value returned by `balanceOf`?

Reply:
You need to either use the original code to figure it out (manually or with the compiler), or trace a call to the function and manually inspect which storage slots are accessed. Neither option is too easy, so for my test I just guessed and verified that the storage value matched the value returned by the `totalSupply` function.